hips/hip-00XX.md

```diff
@@ -0,0 +1,213 @@
+And corresponding API objects field.
+When `true` (default `false`), Helm will enabling override the field conflicts, setting field(s) to the charts values.
+And also make Helm the field manager for the conflicting field(s).
+The `--force-conflicts` flag documentation should explicity mention the downsides of conflict forcing ie. the chart update is overriding object values that another process is expecting to own/manage.
+
+For more details, see the _Overwrite value, become sole manager_ of <https://kubernetes.io/docs/reference/using-api/server-side-apply/#conflicts>
+
+(Of note, Helm upgrade already includes the `--force` flag, which tells Helm to replace objects during upgrade.
+However, given this different semantics of `--force`, a new flag is required)
+
+An additional special case is rollbacks on failed upgrade (`--atomic` flag). It is possible for Helm to attempt to rollback to a version which now has a conflict.
+For which the rollback will fail.
+Since rollbacks are "best effort" and can fail for other reasons as well, it seems best to treat this new rollback failure-case similarly (report to the user via logging and error).
+If the user specified `--force-conflicts` to the upgrade, the corresponding rollback should similarly force conflicts (this is the behavior of `--force`).
+If the rollback is to a release version which previously didn't have SSA enabled, the rollback install should also disable SSA.
+
+#### Special
+
+- `--dry-run=client` won't work with SSA enabled
+- If the user attempts to use SSA with a cluster which does not support it (unlikely: SSA [went GA](https://kubernetes.io/blog/2021/08/06/server-side-apply-ga/) in Kubernetes v1.22), Helm should error
+- Client-side API object validation isn't applicable when SSA enabled (`helm install|upgrade --disable-openapi-validation`)
+- Hooks could be deployed with SSA. But as hook objects are not updated (create/delete only), they are not really affected
+- Stored release manifests should not change (they store the object as Helm intends)
+
+### Implementation
+
+Helm will send object's manifests to the Kubernetes API server via SSA patches. Similar to `kubectl apply`.
+Utilizing the field manager name `"helm"`:
+
+https://github.com/kubernetes/kubectl/blob/197123726db24c61aa0f78d1f0ba6e91a2ec2f35/pkg/cmd/apply/apply.go#L248
+https://github.com/kubernetes/kubectl/blob/197123726db24c61aa0f78d1f0ba6e91a2ec2f35/pkg/cmd/apply/apply.go#L564-L602
+
+## Backwards compatibility
+
+<!-- Describe potential impact and severity on pre-existing code. -->
+
+SSA should be generally compatible functionality-wide with CSA.
+The expectation is that SSA should generally produce equivalent Kubernetes objects as CSA.
+
+There four main behavioural differences for Helm switching to SSA:
+
+1. The differences between object's manifests managed with SSA vs the existing three-way strategic merge process:
+   - Arrays will be merged
+   - Removal of unset fields
+   - Default values
```

Comment:
Which side of the "vs" has which behavior?

Reply:
The handling of removing previously set values in the 3-way merge by setting a null was recently fixed. Do we know how/if SSA will affect that behavior?

Reply:
> Which side of the "vs" has which behavior?

oops, right. these are benefits of SSA. Fixed language.

---

> The handling of removing previously set values in the 3-way merge by setting a null was recently fixed. Do we know how/if SSA will affect that behavior?

This was the input values merge, and dropping fields set to null from input values. This (I guess is kinda the same) but for object manifests applied to the Kubernetes API